src/intervals/trigonometric.jl

```diff
@@ -44,23 +44,25 @@ function sin{T}(a::Interval{T})
     # Different cases depending on the two quadrants:
     if lo_quadrant == hi_quadrant
         a.hi - a.lo > pi_interval(T).lo && return whole_range  # in same quadrant but separated by almost 2pi
-        lo = Interval(sin(a.lo, RoundDown), sin(a.lo, RoundUp))
-        hi = Interval(sin(a.hi, RoundDown), sin(a.hi, RoundUp))
+        lo = @round(sin(a.lo), sin(a.lo)) # Interval(sin(a.lo, RoundDown), sin(a.lo, RoundUp))
+        hi = @round(sin(a.hi), sin(a.hi)) # Interval(sin(a.hi, RoundDown), sin(a.hi, RoundUp))
         return hull(lo, hi)
 
     elseif lo_quadrant==3 && hi_quadrant==0
-        return Interval(sin(a.lo, RoundDown), sin(a.hi, RoundUp))
+        return @round(sin(a.lo), sin(a.hi)) # Interval(sin(a.lo, RoundDown), sin(a.hi, RoundUp))
 
     elseif lo_quadrant==1 && hi_quadrant==2
-        return Interval(sin(a.hi, RoundDown), sin(a.lo, RoundUp))
+        return @round(sin(a.hi), sin(a.lo)) # Interval(sin(a.hi, RoundDown), sin(a.lo, RoundUp))
 
     elseif ( lo_quadrant == 0 || lo_quadrant==3 ) && ( hi_quadrant==1 || hi_quadrant==2 )
-        return Interval(min(sin(a.lo, RoundDown), sin(a.hi, RoundDown)), one(T))
+        return @round(min(sin(a.lo), sin(a.hi)), 1)
+        # Interval(min(sin(a.lo, RoundDown), sin(a.hi, RoundDown)), one(T))
 
     elseif ( lo_quadrant == 1 || lo_quadrant==2 ) && ( hi_quadrant==3 || hi_quadrant==0 )
-        return Interval(-one(T), max(sin(a.lo, RoundUp), sin(a.hi, RoundUp)))
+        return @round(-1, max(sin(a.lo), sin(a.hi)))
```

Comment:
Here the reason is actually different to change this to not use `one(T)`, namely that it was causing problems with the new version of the macro. I now see that these should be solvable; nonetheless, I actually find it much more readable like this, and the same comment about efficiency (that the conversion still happens anyway) applies.